### ISSUE TITLE
RGeo::Error::InvalidGeometry: Unable to cast the geometry to the GEOS Factory

### ISSUE THREAD

**Issue by viktorsmari** [rgeo/rgeo#359]
We have a Rails 7.0.6 app and **all** the tests fail with a RGeo error

`RGeo::Error::InvalidGeometry: Unable to cast the geometry to the GEOS Factory`

Is this related to my localhost setup? How can I debug further?

**Ruby version**:3.0.6
**OS**: Ubuntu 23.04
**geos-config --version**: 3.11.1

```ruby
irb(main):010:0> RGeo::Shapefile::VERSION
=> "3.0.0"
irb(main):011:0> RGeo::VERSION
=> "3.0.0"
irb(main):012:0> RGeo::Geos.version
=> "3.11.1"
```

Gemfile:
```ruby
gem 'activerecord-postgis-adapter'
gem 'rgeo'
gem 'rgeo-geojson'
gem 'rgeo-shapefile'
```
If I comment out the `rgeo*` gems and bundle again, we still get the same error.

### Steps to reproduce
Run `rails test` on any test file, it always shows this error for every test. Even if that test file is not using any RGeo methods.

### Expected behavior
Tests should run?

### Actual behavior
**All** tests error, see error below:

```
RGeo::Error::InvalidGeometry: Unable to cast the geometry to the GEOS Factory
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/geos/capi_factory.rb:272:in `create'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/geos/capi_factory.rb:272:in `polygon'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:321:in `parse_polygon'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:384:in `block in parse_multi_polygon'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:383:in `loop'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:383:in `parse_multi_polygon'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:227:in `parse_type_tag'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:139:in `block in parse'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:123:in `synchronize'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/rgeo-3.0.0/lib/rgeo/wkrep/wkt_parser.rb:123:in `parse'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-postgis-adapter-8.0.2/lib/active_record/connection_adapters/postgis/oid/spatial.rb:89:in `parse_wkt'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-postgis-adapter-8.0.2/lib/active_record/connection_adapters/postgis/oid/spatial.rb:84:in `cast_value'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-postgis-adapter-8.0.2/lib/active_record/connection_adapters/postgis/oid/spatial.rb:70:in `serialize'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:485:in `block (2 levels) in build_fixture_sql'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:482:in `each'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:482:in `map'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:482:in `block in build_fixture_sql'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:474:in `map'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:474:in `build_fixture_sql'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:516:in `block in build_fixture_statements'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:514:in `each'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:514:in `filter_map'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:514:in `build_fixture_statements'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/connection_adapters/abstract/database_statements.rb:402:in `insert_fixtures_set'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/fixtures.rb:630:in `block in insert'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/fixtures.rb:621:in `each'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/fixtures.rb:621:in `insert'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/fixtures.rb:607:in `read_and_insert'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/fixtures.rb:567:in `create_fixtures'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/test_fixtures.rb:275:in `load_fixtures'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/test_fixtures.rb:125:in `setup_fixtures'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activerecord-7.0.6/lib/active_record/test_fixtures.rb:10:in `before_setup'
    /home/viktor/.rbenv/versions/3.0.6/lib/ruby/gems/3.0.0/gems/activesupport-7.0.6/lib/active_support/testing/setup_and_teardown.rb:40:in `before_setup'
```

`dpkg -l | grep geos`
```
ii  libgeos++-dev                                 3.11.1-1                                    amd64        Geometry engine for GIS - C++ development files
ii  libgeos-c1v5:amd64                            3.11.1-1                                    amd64        Geometry engine for Geographic Information Systems - C Library
ii  libgeos-dev                                   3.11.1-1                                    amd64        Geometry engine for GIS - Development files
ii  libgeos-doc                                   3.11.1-1                                    all          Documentation for the GEOS GIS geometry engine library
ii  libgeos3.11.1:amd64                           3.11.1-1                                    amd64        Geometry engine for Geographic Information Systems - C++ Library
```

**Comment by keithdoggett** [comment]
Hi @viktorsmari looks like it's failing while a fixture is being created. Is it possible that you're trying to cast an invalid WKT string to GEOS somewhere while creating test data?

**Comment by viktorsmari** [comment]
Thank you @keithdoggett this was the issue!

```yml
one:
  # before (failed)
  geom: "MULTIPOLYGON (((158.12850950321194 -7.5930849684935575, 158.1108245613315 -7.596477506988496, 158.10755859825898 -7.57808762562538,  154.5151209117339 -5.145125381262376)))"
  # after (works)
  geom: 'POLYGON((-71.456 41.123, -71.454 41.123, -71.454 41.125, -71.456 41.125, -71.456 41.123))'
```

How did you see it was related to a fixture issue? So I can debug myself in the future :nerd_face:

**Comment by keithdoggett** [comment]
@viktorsmari I took a look at the stack trace you sent. The top lines (`serialize` and up) are all postgis adapter/rgeo internal calls that happens when you try to write with WKT. But everything below that is calls from your application, notably most of them contain `fixture` in the call and you mentioned that this happens for every test, so I figured it had to be related to bad data in a fixture.
